client/app/src/Controller/Report/DocumentController.php

```diff
@@ -303,9 +406,12 @@ public function deleteConfirmAction(Request $request, $documentId)
         $report = $document->getReport();
         $fromPage = $request->get('from');
 
-        $backLink = 'summaryPage' == $fromPage
-            ? $this->generateUrl('report_documents_summary', ['reportId' => $report->getId()])
-            : $this->generateUrl('report_documents', ['reportId' => $report->getId()]);
+        if ('reUploadPage' == $fromPage) {
+            $backLink = $this->generateUrl('report_documents_reupload', ['reportId' => $document->getReportId()]);
+        } else {
+            'summaryPage' == $fromPage ? $backLink = $this->generateUrl('report_documents_summary', ['reportId' => $report->getId()])
```

Comment:
You've added another if statement which has lowered the readability of this ternary statement.
I'd suggest moving this to an `elseif`